Signaling/app/src/main/java/com/tokbox/android/tutorials/signaling/MainActivity.java

```diff
@@ -265,11 +187,26 @@ public void onError(Session session, OpentokError opentokError) {
     public void onSignalReceived(Session session, String type, String data, Connection connection) {
 
         boolean remote = !connection.equals(mSession.getConnection());
-        if (type.equals(SIGNAL_TYPE)) {
+        if (SIGNAL_TYPE != null && type.equals(SIGNAL_TYPE)) {
             showMessage(data, remote);
         }
     }
 
+    /* Web Service Coordinator delegate methods */
+
+    @Override
+    public void onSessionConnectionDataReady(String apiKey, String sessionId, String token) {
+
+        Log.d(LOG_TAG, "ApiKey: "+apiKey + " SessionId: "+ sessionId + " Token: "+token);
+
+        initializeSession(apiKey, sessionId, token);
+    }
+
+    @Override
+    public void onWebServiceCoordinatorError(Exception error) {
+        showConfigError("Web Service error", error.getMessage());
```

Comment:
IMHO, we need to call to "finish()" method in this point. As we did for the basic video chat sample:
   Toast.makeText(this, "Web Service error: " + error.getMessage(), Toast.LENGTH_LONG).show();
        finish();